scipy/stats/_qmc.py

```diff
@@ -265,12 +288,13 @@ def discrepancy(sample, iterative=False, method="CD", workers=1):
 
     workers = int(workers)
     if workers == -1:
-        workers = os.cpu_count()
-        if workers is None:
+        cpu_count = os.cpu_count()
```

Comment:
I created here this variable since `os.cpu_count` is returning `Optional[int]` and `workers` is a plain `int`. Since a couple of lines before, I use `workers = int(workers)`, and int is receiving `Union[str, bytes, SupportsInt, _SupportsIndex, _SupportsTrunc]` and not `None`, I couldn't change workers type into `Optional[int]`

Reply:
It was not working to have `workers: Optional[int] = int(workers)`?